### PR TITLE
Add sqloxide

### PR DESCRIPTION
**Pull request by xmnlab**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by xmnlab** [comment]
["  ERROR (sqloxide,lib/python3.10/site-packages/sqloxide/sqloxide.cpython-310-x86_64-linux-gnu.so): .. but ['libgcc-ng'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)"]

**Comment by xmnlab** [comment]
@conda-forge/help-python-c
@conda-forge/help-python this PR is ready for review! thanks!

File: recipes/sqloxide/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    - python
+    - pip
+    - poetry-core >=1.0.0
+    - setuptools >=41.0.0
+    - wheel
+    - setuptools-rust >=0.10.2
+  run:
+    - python
+    - libgcc-ng  # [linux]
+    - compilers
```
Comment (xmnlab) [inline recipes/sqloxide/meta.yaml:29]: maybe this looks very weird here .. but without that, the packaging didn't work (see checks logs)
not sure if would have a better solution for that

File: recipes/sqloxide/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  url: https://github.com/wseaton/{{ name }}/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 5e46a472305d5a5ad1645ff930dc7d8da92563ea5804622c16d11db85f54d4f9
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('rust') }}
```
Comment (isuruf) [inline recipes/sqloxide/meta.yaml:18]:
```suggestion
    - {{ compiler('rust') }}
    - {{ compiler('c') }}
```

File: recipes/sqloxide/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    - python
+    - pip
+    - poetry-core >=1.0.0
+    - setuptools >=41.0.0
+    - wheel
+    - setuptools-rust >=0.10.2
+  run:
+    - python
+    - libgcc-ng  # [linux]
+    - compilers
```
Comment (isuruf) [inline recipes/sqloxide/meta.yaml:30]:
```suggestion
```

**Comment by xmnlab** [comment]
thanks for your suggestion and your review, @isuruf !

**Pull request merged**